src/orbit_determination/gauss_method.jl

```diff
@@ -301,13 +324,225 @@ function gauss_method(observatories::Vector{ObservatoryMPC{T}}, dates::Vector{Da
             g_3 = g_Lagrange(τ_3, r_2)
 
             # Heliocentric velocity of the NEO
-            v_2_vec = (- f_3 * r_vec[1, :] + f_1 * r_vec[3, :]) / (f_1*g_3 - f_3*g_1)
+            v_2_vec = @views (- f_3 * r_vec[1, :] + f_1 * r_vec[3, :]) / (f_1*g_3 - f_3*g_1)
+
+            sol_gauss[i] = GaussSolution{T, U}(
+                vcat(r_vec[2, :], v_2_vec), ρ, D, R_vec,
+                ρ_vec, τ_1, τ_3, f_1, g_1, f_3, g_3
+            )
+        end
+        # Sort solutions by heliocentric range
+        return sort!(sol_gauss, by = x -> norm(cte.(x.statevect[1:3])))
+
+    end
+
+end
+
+@doc raw"""
+    numberofdays(dates::Vector{DateTime})
+    numberofdays(dates::Vector{RadecMPC{T}}) where {T <: AbstractFloat}
+    numberofdays(dates::Vector{Tracklet{T}}) where {T <: AbstractFloat}
+
+Return the time span of `dates` in days. The function assumes `dates` is sorted.
+"""
+numberofdays(dates::Vector{DateTime}) = (dates[end] - dates[1]).value / 86_400_000
+
+function numberofdays(dates::Vector{RadecMPC{T}}) where {T <: AbstractFloat}
+    return (dates[end].date - dates[1].date).value / 86_400_000
+end
+
+function numberofdays(dates::Vector{Tracklet{T}}) where {T <: AbstractFloat}
+    return (dates[end].radec[end].date - dates[1].radec[1].date).value / 86_400_000
+end
+
+@doc raw"""
+    gauss_norm(dates::Vector{DateTime})
+
+Return a measure of how evenly distributed in time a triplet is;
+used within [`gauss_triplets`](@ref) to sort triplets for Gauss method.
+The function assumes `dates` is sorted.
+"""
+gauss_norm(dates::Vector{DateTime}) = abs( (dates[2] - dates[1]).value - (dates[3] - dates[2]).value ) / 86_400_000
 
-            sol_gauss[i] = GaussSolution{T, U}(vcat(r_vec[2, :], v_2_vec), D, R_vec, ρ_vec, τ_1, τ_3, f_1, g_1, f_3, g_3, status[i])
+@doc raw"""
+    gauss_triplets(dates::Vector{DateTime}, max_triplets::Int = 10, max_iter::Int = 100)
+
+Return a vector of `max_triplets` triplets to be used within [`gaussinitcond`](@ref)
+to select the best observations for Gauss method. The triplets are sorted by [`gauss_norm`](@ref).
+"""
+function gauss_triplets(dates::Vector{DateTime}, Δ_min::Period, Δ_max::Period,
+                        avoid::Vector{Vector{Int}}, max_triplets::Int)
+
+    triplets = Vector{Vector{Int}}(undef, 0)
+    L = length(dates)
+    for i_1 in 1:L-2
+        for i_2 in i_1+1:L-1
+            for i_3 in i_2+1:L
+                tmp = [i_1, i_2, i_3]
+                if (Δ_min <= dates[i_3] - dates[i_1] <= Δ_max) && !(tmp in avoid)
+                    push!(triplets, tmp)
+                end
+            end
         end
+    end
 
-        return sort!(sol_gauss, by = x -> norm(x.statevect[1:3]))
+    sort!(triplets, by = x -> gauss_norm(dates[x]))
+
+    n = min(length(triplets), max_triplets)
+
+    return triplets[1:n]
+end
+
+function gauss_triplets(dates::Vector{DateTime}, max_triplets::Int = 10, max_iter::Int = 100)
+    Δ_min = Hour(20)
+    Δ_max = Day(7)
```

Comment:
I'd suggest to keep `Δ_min` and `Δ_max` as keyword arguments in `gauss_triplets`, with default values as they are now

Reply:
Solved by https://github.com/PerezHz/NEOs.jl/pull/31/commits/b67b85ec7f3e4b5d0bf443cce1388194d6fbf464.